### PR TITLE
D3ASIM-3567: Added more supported datetime formats to the str_to_pend…

### PR DESCRIPTION
**Pull request by spyrostz**
…ulum_datetime method.

### CODE REVIEW

**Review by hannesdiedrich — APPROVED** [review]
LGTM

File: d3a_interface/utils.py [hunk 1]
```diff
@@ -185,7 +185,14 @@ def str_to_pendulum_datetime(input_str):
...
             pendulum_time = from_format(input_str, DATE_TIME_FORMAT)
         except ValueError:
-            raise Exception(f"Format is not one of ('{TIME_FORMAT}', '{DATE_TIME_FORMAT}')")
+            try:
+                pendulum_time = from_format(input_str, DATE_TIME_FORMAT_SECONDS)
+            except ValueError:
+                try:
+                    pendulum_time = from_format(input_str, TIME_FORMAT_SECONDS)
+                except ValueError:
+                    raise Exception(f"Format is not one of ('{TIME_FORMAT}', '{DATE_TIME_FORMAT}'"
```
Comment (Hassan754) [inline d3a_interface/utils.py:194]: This is linearly scaling up for each new format we add, I would suggest using a different approach like the following:
```python
supported_formats = iter([TIME_FORMAT, DATE_TIME_FORMAT_SECONDS, TIME_FORMAT_SECONDS])
pendulum_time = None
for format in supported_formats:
    try:
        pendulum_time = from_format(input_str, format)
        break
    except ValueError:
        continue
if not  pendulum_time:
    raise Exception(f"Format is not ....")
```

**Review by Hassan754 — APPROVED** [review]
Left 1 non blocking comment

File: d3a_interface/utils.py (hunk 1 above)
Comment (spyrostz) [reply d3a_interface/utils.py]: Agreed, it is much better this way! Fixing this

**Review by Hassan754 — APPROVED** [review]

File: tests/test_utils.py [hunk 2]
```diff
@@ -51,3 +51,16 @@ def test_convert_datetime_to_ui_str_format(self):
...
+
+    def test_str_to_pendulum_datetime(self):
+        datetime_obj = datetime(year=2021, month=4, day=5, hour=12, minute=30)
+        assert str_to_pendulum_datetime("2021-04-05T12:30") == datetime_obj
+        assert str_to_pendulum_datetime("2021-04-05T12:30:00") == datetime_obj
+        datetime_obj = today(tz="UTC")
+        datetime_obj = datetime_obj.set(hour=12, minute=30)
+        assert str_to_pendulum_datetime("12:30") == datetime_obj
+        assert str_to_pendulum_datetime("12:30:00") == datetime_obj
+        with pytest.raises(Exception):
```
Comment (Hassan754) [inline tests/test_utils.py:63]: Nit, it's better if we explicitly check the `ValueError` exception instead of a broad Exception

Comment (spyrostz) [reply tests/test_utils.py:63]: The code (both original and updated) raise an Exception, not a ValueError. ValueError is the exception that is emitted by from_format and caught in the method

Comment (Hassan754) [reply tests/test_utils.py:63]: Ah you're right, LGTM then

**Review by Hassan754 — APPROVED** [review]

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**